torch/csrc/distributed/autograd/engine/dist_engine.h

```diff
@@ -48,6 +48,10 @@ class TORCH_API DistEngine {
   // Number of backward passes currently running for the Distributed Engine.
   size_t numBackwardPasses() const;
 
+  // Returns key-value pairs consisting of useful debugging information related
+  // to distributed autograd.
+  std::unordered_map<std::string, std::string> getDebugInfo() const;
```

Comment:
This may qualify for getMetrics() instead of getDebugInfo().

Reply:
Although this only has metrics currently, would like to keep this as debug info and str->str so we can easily put additional stuff in here as needed.